### DISCUSSION TITLE
RulesEngine Performance is slow

### DISCUSSION THREAD

**Discussion by lustoro** [microsoft/RulesEngine#538] [Q&A] [answered]
Hello I'm investigating RulesEngine to use it in our project. I've created very simple example where rules count is 70000 and facts count is 6000 and it performs very slow, compilation and execution part tooks over 20 min(without doing post execution actions). here you can check the example - [Rules Engine performance evaluation example](https://github.com/lustoro/RulesEngineExamples/tree/main/RulesEnginePerformanceExample) . I've also  checked [ReSettings](https://github.com/microsoft/RulesEngine/wiki/Getting-Started#resettings) and there is no any setting which could make performance better. I want to check is this expected performance of RulesEngine or there is something I could do in my example to make it perform better? 

Best Regards, 
Lusine

**Comment by abbasc52** [comment, marked as answer]
Hi @lustoro , I am unable to understand your requirement. RulesEngine provides a simple interface to dynamically writing c# rule and evaluating them.

In your case, it looks like you have 70000 rules which is ran against 6000 inputs.

RulesEngine would convert each of these 70K rules into c# delegates and would be evaluated against 6K inputs.
Even without rulesengine running such large number of times would be slow (420,000,000 executions would take couple of minutes). 

RulesEngine can run a workflow with 3 rules in 6 micro sec. That is the performance limit currently. If you could share more details on your requirement. I could help you evaluate whether RulesEngine is good fit for your project

**Comment by lustoro** [comment]
Hi @abbasc52, thanks for your response, mainly want to know performance limitation of RulesEngine. 

Regards,
Lusine
